project/Dependencies.scala

```diff
@@ -109,4 +109,7 @@ object Dependencies {
   val react = "org.webjars" % "react" % "16.5.2"
   val epoch = "org.webjars.npm" % "epoch-charting" % "0.8.4"
   val d3 = "org.webjars.npm" % "d3" % "7.9.0"
+
+  val jacksonModule = "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.17.2"
+  val jacksonDatabind = "com.fasterxml.jackson.core" % "jackson-databind" % "2.17.2"
```

Comment:
👋  @mkurz When I try to bump `net.logstash.logback:logstash-logback-encoder` to the latest version, I'm getting

```com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.14.3 requires Jackson Databind version >= 2.14.0 and < 2.15.0 - Found jackson-databind version 2.17.2```

<details><summary>See full log here</summary>
ch.qos.logback.core.joran.spi.JoranException: Could not open URL [file:/home/runner/work/frontend/frontend/commercial/target/scala-2.13/classes/logback.xml].
11:07:01,193 |-INFO in ch.qos.logback.classic.LoggerContext[frontend-commercial] - This is logback-classic version 1.5.6
	at ch.qos.logback.core.joran.GenericXMLConfigurator.doConfigure(GenericXMLConfigurator.java:70)
11:07:01,195 |-INFO in ch.qos.logback.classic.util.ContextInitializer@7810e5ee - No custom configurators were discovered as a service.
11:07:01,196 |-INFO in ch.qos.logback.classic.util.ContextInitializer@7810e5ee - Trying to configure with ch.qos.logback.classic.joran.SerializedModelConfigurator
	at ch.qos.logback.classic.util.DefaultJoranConfigurator.configureByResource(DefaultJoranConfigurator.java:68)
	at ch.qos.logback.classic.util.DefaultJoranConfigurator.configure(DefaultJoranConfigurator.java:35)
	at ch.qos.logback.classic.util.ContextInitializer.invokeConfigure(ContextInitializer.java:128)
	at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:103)
	at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:66)
	at ch.qos.logback.classic.spi.LogbackServiceProvider.initializeLoggerContext(LogbackServiceProvider.java:52)
	at ch.qos.logback.classic.spi.LogbackServiceProvider.initialize(LogbackServiceProvider.java:41)
	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:196)
	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:183)
	at org.slf4j.LoggerFactory.getProvider(LoggerFactory.java:486)
	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:472)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:421)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:155)
	at org.apache.commons.logging.impl.SLF4JLogFactory.getInstance(SLF4JLogFactory.java:132)
	at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:655)
	at com.gargoylesoftware.htmlunit.javascript.configuration.AbstractJavaScriptConfiguration.<clinit>(AbstractJavaScriptConfiguration.java:53)
	at com.gargoylesoftware.htmlunit.BrowserVersion.initFeatures(BrowserVersion.java:527)
	at com.gargoylesoftware.htmlunit.BrowserVersion.<init>(BrowserVersion.java:[500](https://github.com/guardian/frontend/actions/runs/10524556963/job/29161543366?pr=27315#step:10:501))
	at com.gargoylesoftware.htmlunit.BrowserVersion.<clinit>(BrowserVersion.java:93)
	at test.SingleServerSuite.$init$(package.scala:77)
	at commercial.model.capi.CapiAgentTest.<init>(CapiAgentTest.scala:9)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:454)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:414)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.14.3 requires Jackson Databind version >= 2.14.0 and < 2.15.0 - Found jackson-databind version 2.17.2
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:908)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModules(ObjectMapper.java:1110)
	at com.fasterxml.jackson.databind.ObjectMapper.findAndRegisterModules(ObjectMapper.java:1194)
	at net.logstash.logback.composite.AbstractCompositeJsonFormatter.createJsonFactory(AbstractCompositeJsonFormatter.java:247)
	at net.logstash.logback.composite.AbstractCompositeJsonFormatter.start(AbstractCompositeJsonFormatter.java:117)
	at net.logstash.logback.LogstashFormatter.start(LogstashFormatter.java:144)
	at net.logstash.logback.encoder.CompositeJsonEncoder.start(CompositeJsonEncoder.java:129)
	at ch.qos.logback.core.model.processor.ImplicitModelHandler.postHandleComplex(ImplicitModelHandler.java:207)
	at ch.qos.logback.core.model.processor.ImplicitModelHandler.postHandle(ImplicitModelHandler.java:186)
	at ch.qos.logback.core.model.processor.DefaultProcessor.secondPhaseTraverse(DefaultProcessor.java:257)
	at ch.qos.logback.core.model.processor.DefaultProcessor.secondPhaseTraverse(DefaultProcessor.java:253)
	at ch.qos.logback.core.model.processor.DefaultProcessor.secondPhaseTraverse(DefaultProcessor.java:253)
	at ch.qos.logback.core.model.processor.DefaultProcessor.traversalLoop(DefaultProcessor.java:90)
	at ch.qos.logback.core.model.processor.DefaultProcessor.process(DefaultProcessor.java:106)
	at ch.qos.logback.core.joran.GenericXMLConfigurator.processModel(GenericXMLConfigurator.java:216)
	at ch.qos.logback.core.joran.GenericXMLConfigurator.doConfigure(GenericXMLConfigurator.java:178)
	at ch.qos.logback.core.joran.GenericXMLConfigurator.doConfigure(GenericXMLConfigurator.java:123)
	at ch.qos.logback.core.joran.GenericXMLConfigurator.doConfigure(GenericXMLConfigurator.java:66)
	... 32 more</details>



We're on Play 3.0.5.

I've seen other people having a [similar issue](https://github.com/auth0/jwks-rsa-java/issues/192) and it seems the [solution](https://github.com/auth0/jwks-rsa-java/issues/192#issuecomment-2171800538) was to upgrade everything to latest Jackson. What is your opinion on this? Should we do it? If yes, do you think `jackson-module-scala` and `jackson-databind` enough to avoid runtime errors? These two are enough to run the tests.

Reply:
If you upgrade jackson then you should upgrade all jackson dependencies to the same version, just like recommended in posts you referenced (so not just the scala module and databind).

**However, there might be a nicer solution**: How about to just exclude the jackson transitive dependency from logstash instead?
I have done that [here](https://github.com/playframework/playframework/commit/5d5c6c71c047ccc5499d592169025136f872bbdd) after I ran into the [same problem](https://github.com/playframework/playframework/pull/11861#issuecomment-1970692109).

I think this is safer than to upgrade all jackson dependencies to 2.17 - specially since Play needs to adjust some code for its next major release to correctly support jackson 2.17, see [the build errors we get in this PR](https://github.com/playframework/playframework/pull/12440).

Also, what speaks for just excluding jackson from the logstash dependency is the fact that when jackson was upgraded within logstash, logstash didn't break and didn't need any code modifications:
- 2.14 to 2.15: https://github.com/logfellow/logstash-logback-encoder/commit/f65ce338eac0d021550fb2aa1f80af6843bd800e
- 2.15 to 2.17: https://github.com/logfellow/logstash-logback-encoder/commit/d39941da8fb11bf211756015621ae7a73f03e0cd#diff-860667817975bc227fa0764ac9e23370c9ac901fed13cc61bf4a0580f2413077

That means it should be pretty safe for logstash 8.0 to just continue using Jackson 2.14 that Play ships with.

Reply:
Thanks! That makes sense 👌